tests/integration_tests/conftest.py

```diff
@@ -172,6 +172,26 @@ def _client(request, fixture_utils, session_cloud):
     launch_kwargs = {}
     if name is not None:
         launch_kwargs = {"name": name}
+    if (
```

Comment:
Do you think it makes sense to do image filtering/skipping in the same way and at the same time as we do the [platform filtering/skipping](https://github.com/canonical/cloud-init/blob/master/tests/integration_tests/conftest.py#L39)?

Reply:
Yes, I think it probably does. That'll require some refactoring to drop the dependency on `session_cloud` from the current implementation; we don't have access to that in `pytest_runtest_setup`. I'll see what that looks like and report back.

Reply:
One consequence that occurs to me: this will remove the ability to use the cloud to lookup the image's OS and release. LXD images from streams, for example, do have this metadata:

```
$ lxc image show ubuntu:ffae848ee5a0
auto_update: false
properties:
  architecture: amd64
  description: ubuntu 20.04 LTS amd64 (release) (20200529.1)
  label: release
  os: ubuntu
  release: focal
  serial: "20200529.1"
  type: root.tar.xz
  version: "20.04"
public: true
expires_at: 2025-04-23T00:00:00Z
profiles: []
```

I don't know that this is a particularly huge loss; I suspect that such reverse lookups will not be generically implementable anyway (on EC2, for example, we'd be parsing something like "ubuntu/images-testing/hvm-ssd/ubuntu-trusty-daily-amd64-server-20191107"), so the odds of them being used would be pretty low.